weave/trace_server/trace_server_interface.py

```diff
@@ -318,6 +319,75 @@ class FileContentReadRes(BaseModel):
     content: bytes
 
 
+class FeedbackPayloadReactionReq(BaseModel):
+    emoji: str
+
+
+class FeedbackPayloadNoteReq(BaseModel):
+    note: str = Field(min_length=1, max_length=1024)
+
+
+class FeedbackCreateReq(BaseModel):
+    project_id: str = Field(examples=["entity/project"])
+    weave_ref: str = Field(examples=["weave:///entity/project/object/name:digest"])
+    creator: typing.Optional[str] = Field(default=None, examples=["Jane Smith"])
+    feedback_type: str = Field(examples=["custom"])
+    payload: typing.Dict[str, typing.Any] = Field(
+        examples=[
+            {
+                "key": "value",
+            }
+        ]
+    )
+
+
+class FeedbackCreateReqForInsert(FeedbackCreateReq):
+    wb_user_id: str
+
+
+# The response provides the additional fields needed to convert a request
+# into a complete Feedback.
+class FeedbackCreateRes(BaseModel):
+    id: str
+    created_at: datetime.datetime
+    wb_user_id: str
+    payload: typing.Dict[str, typing.Any]  # If not empty, replace payload
+
+
+class Feedback(FeedbackCreateReqForInsert):
+    id: str
+    created_at: datetime.datetime
+
+
+class FeedbackQueryReq(BaseModel):
+    project_id: str = Field(examples=["entity/project"])
+    fields: typing.Optional[list[str]] = Field(
+        default=None, examples=[["id", "feedback_type", "payload.note", "count(*)"]]
+    )
+    query: typing.Optional[Query] = None
+    # TODO: I think I would prefer to call this order_by to match SQL, but this is what calls API uses
+    # TODO: Might be nice to have shortcut for single field and implied ASC direction
+    # TODO: I think _SortBy shouldn't have leading underscore
+    sort_by: typing.Optional[typing.List[_SortBy]] = None
+    limit: typing.Optional[int] = Field(default=None, examples=[10])
+    offset: typing.Optional[int] = Field(default=None, examples=[0])
+
+
+class FeedbackQueryRes(BaseModel):
+    # Note: this is not a list of Feedback because user can request any fields.
+    result: list[dict[str, typing.Any]]
+
+
+class FeedbackPurgeReq(BaseModel):
+    project_id: str = Field(examples=["entity/project"])
+    # TODO: Should we make this required?
+    query: typing.Optional[Query] = None
```

Comment:
I still don't like this ... feels very easy to mistakenly delete everything.

Reply:
Opinion on how I should mitigate that? Add a limit on how many can rows be deleted at once? Change the API to only allow deletion by ID? Keep it as a (required) Query parameter but raise if it is too generic (by some heuristic)?

Reply:
If you want to retain the query capability, I'd prefer to just allow a single query shape that matches what you need (which i think could just be an "and" over ids)